packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/RealmAny.kt

```diff
@@ -0,0 +1,335 @@
+     * safely.
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `Long`.
+     */
+    public fun asLong(): Long
+
+    /**
+     * Returns the value from this `RealmAny` as a [Boolean].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `Boolean`.
+     */
+    public fun asBoolean(): Boolean
+
+    /**
+     * Returns the value from this `RealmAny` as a [String].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `String`.
+     */
+    public fun asString(): String
+
+    /**
+     * Returns the value from this `RealmAny` as a [Float].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `Float`.
+     */
+    public fun asFloat(): Float
+
+    /**
+     * Returns the value from this `RealmAny` as a [Double].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `Double`.
+     */
+    public fun asDouble(): Double
+
+    /**
+     * Returns the value from this `RealmAny` as an [ObjectId].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `ObjectId`.
+     */
+    @Deprecated("Use the BSON ObjectId variant instead", ReplaceWith("RealmAny.asObjectId"))
+    public fun asRealmObjectId(): ObjectId
+
+    /**
+     * Returns the value from this `RealmAny` as a [BsonObjectId].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to
+     * `BsonObjectId`.
+     */
+    public fun asObjectId(): BsonObjectId
+
+    /**
+     * Returns the value from this `RealmAny` as a [ByteArray].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to
+     * `ByteArray`.
+     */
+    public fun asByteArray(): ByteArray
+
+    /**
+     * Returns the value from this `RealmAny` as a [RealmInstant].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to
+     * `RealmInstant`.
+     */
+    public fun asRealmInstant(): RealmInstant
+
+    /**
+     * Returns the value from this `RealmAny` as a [RealmUUID].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to
+     * `RealmUUID`.
+     */
+    public fun asRealmUUID(): RealmUUID
+
+    /**
+     * Returns the value from this RealmAny as a [RealmObject] of type [T].
+     * @throws [IllegalStateException] if the stored value cannot be safely converted to `T`.
+     */
+    public fun <T : RealmObject> asRealmObject(clazz: KClass<T>): T
+
+    /**
+     * Two [RealmAny] instances are equal if and only if their types and contents are the equal.
+     */
+    override fun equals(other: Any?): Boolean
+
+    override fun hashCode(): Int
+
+    public companion object {
+        /**
+         * Creates an unmanaged `RealmAny` instance from a [Short] value.
+         */
+        public fun create(value: Short): RealmAny =
```

Comment:
Rereading #835 I am not sure we caught the conclusion, but ... 🤔 should we use `from(...)` to be consistent with `RealmInstant.from()`, etc.

Reply:
I will go for create unless @cmelchior changes opinion during review.